plugin/src/sidebar.ts

```diff
@@ -35,9 +36,15 @@ reearth.on("message", ({ action, payload }: PostMessageProps) => {
       },
     });
     reearth.clientStorage.setAsync("overrides", payload);
+    reearth.clientStorage.getAsync("doNotShowWelcome").then((value: any) => {
+      doNotShowWelcome = value;
+    });
```

Comment:
The `getAsync` check is good here on `initSidebar`, but inside the `then()` you should be doing a bit more.
Here you should check if doNotShowWelcome is truthy, and if so, do nothing. If it is falsey (undefined or false), this is where you should do the `reearth.modal.show(welcomeScreenHtml, ....)`

Reply:
Also, I don't think you'll need the `let doNotShowWelcome = false;` on line 17 anymore since checking the storage is enough

Reply:
done

Reply:
2e7e5ada980318f9de64a960bc7f473718816a86